src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/StartupObjectsEnumProvider.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    /// Returns the set of Startup objects (or entry point types) in a project.
+    /// </summary>
+    [ExportDynamicEnumValuesProvider("StartupObjectsEnumProvider")]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class StartupObjectsEnumProvider : IDynamicEnumValuesProvider
+    {
+        private readonly Workspace _workspace;
+        private readonly UnconfiguredProject _unconfiguredProject;
+
+        [ImportingConstructor]
+        public StartupObjectsEnumProvider([Import(typeof(VisualStudioWorkspace))] Workspace workspace, UnconfiguredProject unconfiguredProject)
+        {
+            Requires.NotNull(workspace, nameof(workspace));
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+
+            _workspace = workspace;
+            _unconfiguredProject = unconfiguredProject;
+        }
+
+        public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair> options)
+        {
+            return Task.FromResult<IDynamicEnumValuesGenerator>(new StartupObjectsEnumGenerator(_workspace, _unconfiguredProject));
+        }
+    }
+
+    internal class StartupObjectsEnumGenerator : IDynamicEnumValuesGenerator
```

Comment:
The pattern with the enumproviders in the repo is to keep the generator as a nested class in the same file. Being consistent here.